spec/compiler/compiler_spec.cr

```diff
@@ -28,6 +28,24 @@ describe "Compiler" do
     end
   end
 
+  it "outputs correct cc command in preference to a filename when cross compile" do
+    crystal_bin = File.realpath(ENV["CRYSTAL_SPEC_COMPILER_BIN"]? || "bin/crystal")
+    Dir.cd compiler_datapath do
+      with_temp_executable "compiler_spec_output" do |path|
+        FileUtils.mkdir_p("#{path}/bin")
+        expected_bin = "#{path}/bin/compiler_sample-linux-x86_64"
+        output = IO::Memory.new
+        Process.run(
+          crystal_bin,
+          ["build"].concat(program_flags_options).concat(["--cross-compile", "--target=x86_64-unknown-linux-gnu", "-o", expected_bin, "compiler_sample"]),
+          output: output
+        )
+        File.exists?("#{expected_bin}.o").should be_true
+        output.to_s.should contain("cc #{expected_bin}.o -o #{expected_bin} ")
```

Comment:
```suggestion
        output.to_s.should contain(" #{expected_bin}.o -o #{expected_bin} ")
```

Alternatively the linker could also be expressed fully as `ENV["CC"]? || {{ flag?(:msvc) ? "cl.exe" : "cc" }}`.

Reply:
What about the `.o` extension, is it the same on windows?

Reply:
That too, it's `.obj`

Reply:
Currently the spec is cross-compiling explicitly to `x86_64-unknown-linux-gnu`, so no need to cover windows. (It would be a good enhancement for better coverage, though).

Reply:
@HertzDevil Actually, I don't think the linker expression is correct. Remember the cross-compile target is linux. So the linker could never be `cl.exe`, even if the host compiler runs on Windows.

Reply:
I guess it could still be `ENV["CC"]?`?